packages/block-library/src/navigation/edit/index.js

```diff
@@ -213,6 +219,12 @@ function Navigation( {
 		hasResolvedCanUserCreateNavigationMenu,
 	} = useNavigationMenu( ref );
 
+	const [ navigationInnerBlocks, onInput, onChange ] = useEntityBlockEditor(
+		'postType',
+		'wp_navigation',
+		{ id: ref }
+	);
```

Comment:
Is `useEntityBlockEditor` and the `BlockEditorProvider` definitely needed?

Normal list view works fine for editing navigation inner blocks, and doesn't have this.

If this is copied from the navigation sidebar, I believe it is required there because it's possible to show a menu that isn't within the editor content (the blocks aren't in the block editor's store). 

My understanding is that isn't the possible here, so I think it may not be needed.

Reply:
This _is_ a quick late in the day PR, should have made it a draft. However, without using a block entity provider, selecting an item from ListView selects the block in the canvas. 🤷🏻

Reply:
> However, without using a block entity provider, selecting an item from ListView selects the block in the canvas. 🤷🏻

That's intentional, because List View is a way to select blocks in the canvas.

The reason why BlockEditorProvider would have this effect is because it essentially creates an entirely separate block editor store, with different selection state to the main block editor.

For this new List View in the nav block, what should selecting a block in List View do?

Reply:
Nothing, it should select the item _in the list view_ so users can drag/drop/use the more menu to uodate the navigation in a tree view.

Reply:
> Nothing, it should select the item in the list view so users can drag/drop/use the more menu to uodate the navigation in a tree view.

Selecting in List View doesn't really mean anything though. Drag and drop and the more menu are usable when a block isn't selected.

I think the main thing is that this is going to be a confusing accessibility situation, since this List View contains buttons that say they will select the block, but they actually do nothing.

My opinion is that the best thing would be to not use `BlockEditorProvider` for this purpose, but find solutions within List View for this problem.

Reply:
Made a todo to give up using BlockEditorProvider 👌🏻